httprule/client.go

```diff
@@ -43,6 +95,13 @@ func (c *ClientConn) Invoke(ctx context.Context, method string, args, reply inte
 	if err != nil {
 		return err
 	}
+	if c.header != nil {
+		for key, vals := range c.header {
+			for _, v := range vals {
+				req.Header.Add(key, v)
```

Comment:
have you considered adding this to the client.Transport instead so I would also be captured for the streaming (iff we ever implement that)?

Reply:
Yes. I did consider that. Then I read this comment on the `RoundTripper` interface:

> RoundTrip should not attempt to handle higher-level protocol details such as redirects, authentication, or cookies...

That thing we did in vcards apparently was not the right way to do this. This idea of client "middleware" wrapping the roundtripper goes against the docs.

Reply:
I guess adding headers may not fall into those categories. At the time I was thinking of only adding the auth header via the RoundTripper if i got an auth required response.